# Importan Note untuk dibaca

<li><b>setelah april 2022 (2022_04) persentase untuk PPN adalah menjadi 11% </b>

# Import File dan Change Header

## Importing Library

In [1]:
import pandas as pd
import seaborn as sns
import datetime

## Tahun 2021

In [2]:
header = ["Tanggal","Product_ID","Product","Location_ID","Location","Quantity","Price_AfterPPN","Price_BeforePPN"]

januari2021 = pd.read_csv("sellout_watsons_2021_01.csv",header =None)
januari2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

februari2021 = pd.read_csv("sellout_watsons_2021_02.csv",header =None)
februari2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

maret2021 = pd.read_csv("sellout_watsons_2021_03.csv",header =None)
maret2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

april2021 = pd.read_csv("sellout_watsons_2021_04.csv",header =None)
april2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

mei2021 = pd.read_csv("sellout_watsons_2021_05.csv",header =None)
mei2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

juni2021 = pd.read_csv("sellout_watsons_2021_06.csv",header =None)
juni2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

juli2021 = pd.read_csv("sellout_watsons_2021_07.csv",header =None)
juli2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

agustus2021 = pd.read_csv("sellout_watsons_2021_08.csv",header =None)
agustus2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

september2021 = pd.read_csv("sellout_watsons_2021_09.csv",header =None)
september2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

oktober2021 = pd.read_csv("sellout_watsons_2021_10.csv",header =None)
oktober2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

november2021 = pd.read_csv("sellout_watsons_2021_11.csv",header =None)
november2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

desember2021 = pd.read_csv("sellout_watsons_2021_12.csv",header =None)
desember2021.columns = header # pergantian header didapatkan setelah dicheck melalui data

## Tahun 2022

In [3]:
januari2022 = pd.read_csv("sellout_watsons_2022_01.csv",header =None)
januari2022.columns = header # pergantian header didapatkan setelah dicheck melalui data

februari2022 = pd.read_csv("sellout_watsons_2022_02.csv",header =None)
februari2022.columns = header # pergantian header didapatkan setelah dicheck melalui data

maret2022 = pd.read_csv("sellout_watsons_2022_03.csv",header =None)
maret2022.columns = header # pergantian header didapatkan setelah dicheck melalui data

april2022 = pd.read_csv("sellout_watsons_2022_04.csv",header =None)
april2022.columns = header # pergantian header didapatkan setelah dicheck melalui data

mei2022 = pd.read_csv("sellout_watsons_2022_05.csv",header =None)
mei2022.columns = header # pergantian header didapatkan setelah dicheck melalui data

# Configurasi Tanggal menggunakan library

In [4]:
maret2021.head(5)

Tanggal  Product_ID                         Product  Location_ID  \
0  2021-02-22       16252   WARDAH LIGHTENING OIL-INFUSED            1   
1  2021-02-22       16392  EMINA LIP GLOSSY STAIN 02 APPL            1   
2  2021-02-22       16394  EMINA LIP GLOSSY STAIN 04 PEAC            1   
3  2021-02-22       16884  WARDAH LIGHTENING CLEANSING MI            1   
4  2021-02-22       18040       KAHF FW BRIGHTENING 100ML            1   

              Location  Quantity  Price_AfterPPN  Price_BeforePPN  
0  PONDOK INDAH MALL 2         1         21000.0         19090.91  
1  PONDOK INDAH MALL 2         1         38000.0         34545.45  
2  PONDOK INDAH MALL 2         1         38000.0         34545.45  
3  PONDOK INDAH MALL 2         1         16000.0         14545.45  
4  PONDOK INDAH MALL 2         1         36900.0         33545.45

In [5]:
januari2021.head(5)

Tanggal  Product_ID                         Product  Location_ID  \
0  12/28/2020       10568  EMINA CREAMYTINT 01 BRICK TOWN            1   
1  12/28/2020       15002   MO P/STAY TRANSPROOF LIPS CRM            1   
2  12/28/2020       15018  EMINA MAGIC POTION 04 SIENNA 5            1   
3  12/28/2020       15788  WARDAH COLORFIT VEL LIP/MAT 12            1   
4  12/28/2020       16288  WARDAH A/D PORE BLACKHEAD BALM            1   

              Location  Quantity  Price_AfterPPN  Price_BeforePPN  
0  PONDOK INDAH MALL 2         1           44000         40000.00  
1  PONDOK INDAH MALL 2         1          120000        109090.91  
2  PONDOK INDAH MALL 2         1           36800         33454.55  
3  PONDOK INDAH MALL 2         1           58400         53090.91  
4  PONDOK INDAH MALL 2         1           33500         30454.55

<b>pada januari 2021, format tanggal yang digunakan berbeda</b>

In [6]:
def convert(tanggal):
    return datetime.datetime.strptime(tanggal[0],"%m/%d/%Y").strftime("%Y-%m-%d")

In [7]:
januari2021.Tanggal = januari2021.apply(convert,axis=1)

In [8]:
januari2021.head(5)

Tanggal  Product_ID                         Product  Location_ID  \
0  2020-12-28       10568  EMINA CREAMYTINT 01 BRICK TOWN            1   
1  2020-12-28       15002   MO P/STAY TRANSPROOF LIPS CRM            1   
2  2020-12-28       15018  EMINA MAGIC POTION 04 SIENNA 5            1   
3  2020-12-28       15788  WARDAH COLORFIT VEL LIP/MAT 12            1   
4  2020-12-28       16288  WARDAH A/D PORE BLACKHEAD BALM            1   

              Location  Quantity  Price_AfterPPN  Price_BeforePPN  
0  PONDOK INDAH MALL 2         1           44000         40000.00  
1  PONDOK INDAH MALL 2         1          120000        109090.91  
2  PONDOK INDAH MALL 2         1           36800         33454.55  
3  PONDOK INDAH MALL 2         1           58400         53090.91  
4  PONDOK INDAH MALL 2         1           33500         30454.55

<b> semua tabel telah memiliki format yang sama <b>

# Checking Harga Normal Price apakah selalu konstan

In [9]:
januari2021_PIM = januari2021[januari2021.Location_ID == 1]

In [10]:
januari2021_PIM.Product_ID.value_counts().head(5)

16391    19
83179    19
55408    17
18208    15
19170    14
Name: Product_ID, dtype: int64

In [11]:
januari2021[januari2021.Product_ID == 16391]

Tanggal  Product_ID                         Product  Location_ID  \
5      2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
59     2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            2   
429    2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU           22   
476    2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU           23   
555    2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU           24   
...           ...         ...                             ...          ...   
95517  2021-01-24       16391  EMINA LIP GLOSSY STAIN 01 AUTU          149   
95608  2021-01-24       16391  EMINA LIP GLOSSY STAIN 01 AUTU          153   
95647  2021-01-24       16391  EMINA LIP GLOSSY STAIN 01 AUTU          154   
95673  2021-01-24       16391  EMINA LIP GLOSSY STAIN 01 AUTU          155   
95807  2021-01-24       16391  EMINA LIP GLOSSY STAIN 01 AUTU          162   

                   Location  Quantity  Price_AfterPPN  Price_BeforePPN  
5       PONDOK INDAH MALL 2         5          237500        215909.09  
59     MALL KELAPA GADING 3         3          142500        129545.45  
429       PLAZA PONDOK GEDE         1           47500         43181.82  
476       SUMMARECON BEKASI         6          285000        259090.91  
555     CITRA GRAND CIBUBUR         1           47500         43181.82  
...                     ...       ...             ...              ...  
95517   THEE MATIC MAJALAYA         1           33250         30227.27  
95608        NEW MARGO CITY         4          133000        120909.09  
95647      RAMAYANA MAKASAR         1           33250         30227.27  
95673    PANAKUKANG MAKASAR         2           66500         60454.55  
95807  THEPARK MAL SAWANGAN         3           99750         90681.82  

[938 rows x 8 columns]

In [12]:
test = januari2021_PIM[januari2021_PIM.Product_ID == 16391].Price_AfterPPN / januari2021_PIM[januari2021_PIM.Product_ID == 16391].Quantity

In [13]:
test.value_counts()

47500.0    11
33250.0     8
dtype: int64

In [14]:
januari2021_PIM[januari2021_PIM.Product_ID == 16391]

Tanggal  Product_ID                         Product  Location_ID  \
5      2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
3707   2020-12-29       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
11646  2020-12-31       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
15961  2021-01-01       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
26109  2021-01-03       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
30596  2021-01-04       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
33550  2021-01-05       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
36322  2021-01-06       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
39128  2021-01-07       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
45446  2021-01-09       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
49940  2021-01-10       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
57254  2021-01-12       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
59512  2021-01-13       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
67176  2021-01-16       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
71442  2021-01-17       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
76012  2021-01-18       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
82552  2021-01-21       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
84785  2021-01-22       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
87434  2021-01-23       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   

                  Location  Quantity  Price_AfterPPN  Price_BeforePPN  
5      PONDOK INDAH MALL 2         5          237500        215909.09  
3707   PONDOK INDAH MALL 2         4          190000        172727.27  
11646  PONDOK INDAH MALL 2         2           95000         86363.64  
15961  PONDOK INDAH MALL 2         1           47500         43181.82  
26109  PONDOK INDAH MALL 2         2           95000         86363.64  
30596  PONDOK INDAH MALL 2         1           47500         43181.82  
33550  PONDOK INDAH MALL 2         1           47500         43181.82  
36322  PONDOK INDAH MALL 2         1           47500         43181.82  
39128  PONDOK INDAH MALL 2         3           99750         90681.82  
45446  PONDOK INDAH MALL 2         3           99750         90681.82  
49940  PONDOK INDAH MALL 2         5          166250        151136.36  
57254  PONDOK INDAH MALL 2         1           47500         43181.82  
59512  PONDOK INDAH MALL 2         1           47500         43181.82  
67176  PONDOK INDAH MALL 2         1           33250         30227.27  
71442  PONDOK INDAH MALL 2         1           33250         30227.27  
76012  PONDOK INDAH MALL 2         2           95000         86363.64  
82552  PONDOK INDAH MALL 2         1           33250         30227.27  
84785  PONDOK INDAH MALL 2         2           66500         60454.55  
87434  PONDOK INDAH MALL 2         2           66500         60454.55

In [15]:
januari2021_PIM[januari2021_PIM.Product_ID == 16391][test == 33250].head(3)

Tanggal  Product_ID                         Product  Location_ID  \
39128  2021-01-07       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
45446  2021-01-09       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
49940  2021-01-10       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   

                  Location  Quantity  Price_AfterPPN  Price_BeforePPN  
39128  PONDOK INDAH MALL 2         3           99750         90681.82  
45446  PONDOK INDAH MALL 2         3           99750         90681.82  
49940  PONDOK INDAH MALL 2         5          166250        151136.36

In [16]:
januari2021_PIM[januari2021_PIM.Product_ID == 16391][test == 47500].head(3)

Tanggal  Product_ID                         Product  Location_ID  \
5      2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
3707   2020-12-29       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
11646  2020-12-31       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   

                  Location  Quantity  Price_AfterPPN  Price_BeforePPN  
5      PONDOK INDAH MALL 2         5          237500        215909.09  
3707   PONDOK INDAH MALL 2         4          190000        172727.27  
11646  PONDOK INDAH MALL 2         2           95000         86363.64

In [17]:
januari2021_PIM[januari2021_PIM.Product_ID == 16391].head(3)

Tanggal  Product_ID                         Product  Location_ID  \
5      2020-12-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
3707   2020-12-29       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
11646  2020-12-31       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   

                  Location  Quantity  Price_AfterPPN  Price_BeforePPN  
5      PONDOK INDAH MALL 2         5          237500        215909.09  
3707   PONDOK INDAH MALL 2         4          190000        172727.27  
11646  PONDOK INDAH MALL 2         2           95000         86363.64

<ul>
    <li> harga prodak tidak konstan karena terdapat promo pada pembelian in store
    <li> dan bisa saja untuk bulan yang berbeda, harga product juga meningkat
</ul>

# Check Pergantian PP

In [18]:
april2022[april2022.Quantity == 1].head(3)

Tanggal  Product_ID                         Product  Location_ID  \
0  2022-04-04       10568  EMINA CREAMYTINT 01 BRICK TOWN            1   
1  2022-04-04       14649            WARDAH COND HF 170ML            1   
2  2022-04-04       15000   MO P/STAY TRANSPROOF LIPS CRM            1   

              Location  Quantity  Price_AfterPPN  Price_BeforePPN  
0  PONDOK INDAH MALL 2         1         45000.0         40540.54  
1  PONDOK INDAH MALL 2         1         25800.0         23243.24  
2  PONDOK INDAH MALL 2         1        102400.0         92252.25

In [19]:
(april2022.Price_AfterPPN - april2022.Price_BeforePPN)/ april2022.Price_BeforePPN

0         0.110000
1         0.110000
2         0.110000
3         0.110000
4         0.110000
            ...   
158165    0.109999
158166    0.110000
158167    0.110000
158168    0.110000
158169    0.110000
Length: 158170, dtype: float64

<b> pergantian Pajak Prodak dari 10% menjadi 11% terbukti dan telah terjadi </b>

# Apakah harga untuk satu prodak berbeda untuk store yang berbeda

In [20]:
maret2022

Tanggal  Product_ID                         Product  Location_ID  \
0       2022-02-28       10572  EMINA CREAMYTINT 05 CHERRY SOD            1   
1       2022-02-28       11554   Crystallure Supr Dbl Act Mclr            1   
2       2022-02-28       15018  EMINA MAGIC POTION 04 SIENNA 5            1   
3       2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
4       2022-02-28       16392  EMINA LIP GLOSSY STAIN 02 APPL            1   
...            ...         ...                             ...          ...   
149361  2022-04-03       55271  WARDAH C/S FOAMING CLEANSER 10          917   
149362  2022-04-03       55367  WARDAH C/S BRIGHT DAY CREAM 15          917   
149363  2022-04-03       66203  WARDAH RENEW YOU NGHT CREM 30G          917   
149364  2022-04-03       66204  WARDAH RENEW YOU DAY CREAM 30G          917   
149365  2022-04-03       80429   EMINA DAILY MAT LOOSE PWDR 02          917   

                   Location  Quantity  Price_AfterPPN  Price_BeforePPN  
0       PONDOK INDAH MALL 2         1         35200.0         32000.00  
1       PONDOK INDAH MALL 2         1        123200.0        112000.00  
2       PONDOK INDAH MALL 2         1         36800.0         33454.55  
3       PONDOK INDAH MALL 2         5        190000.0        172727.27  
4       PONDOK INDAH MALL 2         1         38000.0         34545.45  
...                     ...       ...             ...              ...  
149361  ECOMM CASH DELIVERY         4         81959.0         73836.94  
149362  ECOMM CASH DELIVERY         2         61085.0         55031.53  
149363  ECOMM CASH DELIVERY         2        130400.0        117477.48  
149364  ECOMM CASH DELIVERY         1         48700.0         43873.87  
149365  ECOMM CASH DELIVERY         1         22041.0         19856.76  

[149366 rows x 8 columns]

In [21]:
maret2022[maret2022.Product_ID == 16391]

Tanggal  Product_ID                         Product  Location_ID  \
3       2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
112     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            2   
252     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            8   
351     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU           10   
433     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU           12   
...            ...         ...                             ...          ...   
148975  2022-04-03       16391  EMINA LIP GLOSSY STAIN 01 AUTU          807   
149057  2022-04-03       16391  EMINA LIP GLOSSY STAIN 01 AUTU          808   
149258  2022-04-03       16391  EMINA LIP GLOSSY STAIN 01 AUTU          813   
149331  2022-04-03       16391  EMINA LIP GLOSSY STAIN 01 AUTU          915   
149357  2022-04-03       16391  EMINA LIP GLOSSY STAIN 01 AUTU          917   

                    Location  Quantity  Price_AfterPPN  Price_BeforePPN  
3        PONDOK INDAH MALL 2         5       190000.00        172727.27  
112     MALL KELAPA GADING 3         5       190000.00        172727.27  
252          KOTA KASABLANKA         1        38000.00         34545.45  
351         MAL ARTHA GADING         1        38000.00         34545.45  
433              KRAMAT JATI         2        76000.00         69090.91  
...                      ...       ...             ...              ...  
148975     LAZADA MARGO CITY         1        35600.00         32072.07  
149057        SHOPEE WATSONS        46      1559283.33       1404759.76  
149258             TOKOPEDIA         1        47500.00         42792.79  
149331   ECOMM HOME DELIVERY         3        92422.00         83263.06  
149357   ECOMM CASH DELIVERY         1         3950.00          3558.56  

[1691 rows x 8 columns]

In [22]:
maret2022["Harga_Satuan"] = maret2022.Price_AfterPPN/ maret2022.Quantity

In [23]:
maret2022[maret2022.Product_ID == 16391].Harga_Satuan.sort_values().value_counts()

38000.000000    1119
33250.000000     251
47500.000000     136
48500.000000     105
33333.000000      29
33333.333333       7
40916.500000       6
38388.666667       5
40300.000000       3
35600.000000       2
36366.400000       2
44708.250000       1
38052.272727       1
38055.333333       1
40375.000000       1
39000.000000       1
46444.444444       1
40416.500000       1
37568.181818       1
42750.000000       1
40533.333333       1
3950.000000        1
37125.000000       1
36703.555556       1
8000.000000        1
35687.444444       1
35361.111111       1
35252.000000       1
34928.571429       1
34833.333333       1
33950.000000       1
33897.463696       1
33479.838710       1
32701.000000       1
32353.000000       1
30807.333333       1
36366.600000       1
Name: Harga_Satuan, dtype: int64

In [24]:
maret2022[(maret2022.Harga_Satuan == 38000) & (maret2022.Product_ID == 16391)]

Tanggal  Product_ID                         Product  Location_ID  \
3       2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            1   
112     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            2   
252     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU            8   
351     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU           10   
433     2022-02-28       16391  EMINA LIP GLOSSY STAIN 01 AUTU           12   
...            ...         ...                             ...          ...   
96050   2022-03-23       16391  EMINA LIP GLOSSY STAIN 01 AUTU          181   
96167   2022-03-23       16391  EMINA LIP GLOSSY STAIN 01 AUTU          807   
96250   2022-03-23       16391  EMINA LIP GLOSSY STAIN 01 AUTU          808   
96393   2022-03-23       16391  EMINA LIP GLOSSY STAIN 01 AUTU          813   
100296  2022-03-24       16391  EMINA LIP GLOSSY STAIN 01 AUTU          813   

                    Location  Quantity  Price_AfterPPN  Price_BeforePPN  \
3        PONDOK INDAH MALL 2         5        190000.0        172727.27   
112     MALL KELAPA GADING 3         5        190000.0        172727.27   
252          KOTA KASABLANKA         1         38000.0         34545.45   
351         MAL ARTHA GADING         1         38000.0         34545.45   
433              KRAMAT JATI         2         76000.0         69090.91   
...                      ...       ...             ...              ...   
96050     GRAGE MALL CIREBON         1         38000.0         34545.45   
96167      LAZADA MARGO CITY         1         38000.0         34545.45   
96250         SHOPEE WATSONS        17        646000.0        587272.73   
96393              TOKOPEDIA         1         38000.0         34545.45   
100296             TOKOPEDIA         3        114000.0        103636.36   

        Harga_Satuan  
3            38000.0  
112          38000.0  
252          38000.0  
351          38000.0  
433          38000.0  
...              ...  
96050        38000.0  
96167        38000.0  
96250        38000.0  
96393        38000.0  
100296       38000.0  

[1119 rows x 9 columns]

In [25]:
maret2022[maret2022.Harga_Satuan == 47500]

Tanggal  Product_ID                         Product  Location_ID  \
64      2022-02-28       67479  WARDAH E.SHADOW N.CLOR PSIONTE            1   
298     2022-02-28       67478   WARDAH E.SHADOW N.CLOR CLASIC            8   
417     2022-02-28       67478   WARDAH E.SHADOW N.CLOR CLASIC           11   
455     2022-02-28       68150  WARDAH RENEW A/AGING D/CMSPF30           12   
479     2022-02-28       67478   WARDAH E.SHADOW N.CLOR CLASIC           13   
...            ...         ...                             ...          ...   
144964  2022-04-02       17104    EMINA CREAMATTE 18 ONDE-ONDE          807   
145317  2022-04-02       16391  EMINA LIP GLOSSY STAIN 01 AUTU          813   
148977  2022-04-03       16395  EMINA LIP GLOSSY STAIN 05 SPRI          807   
148980  2022-04-03       17104    EMINA CREAMATTE 18 ONDE-ONDE          807   
149258  2022-04-03       16391  EMINA LIP GLOSSY STAIN 01 AUTU          813   

                   Location  Quantity  Price_AfterPPN  Price_BeforePPN  \
64      PONDOK INDAH MALL 2         1         47500.0         43181.82   
298         KOTA KASABLANKA         2         95000.0         86363.64   
417            PLAZA BUARAN         1         47500.0         43181.82   
455             KRAMAT JATI         1         47500.0         43181.82   
479         LOKASARI SQUARE         1         47500.0         43181.82   
...                     ...       ...             ...              ...   
144964    LAZADA MARGO CITY         1         47500.0         42792.79   
145317            TOKOPEDIA         2         95000.0         85585.59   
148977    LAZADA MARGO CITY         1         47500.0         42792.79   
148980    LAZADA MARGO CITY         1         47500.0         42792.79   
149258            TOKOPEDIA         1         47500.0         42792.79   

        Harga_Satuan  
64           47500.0  
298          47500.0  
417          47500.0  
455          47500.0  
479          47500.0  
...              ...  
144964       47500.0  
145317       47500.0  
148977       47500.0  
148980       47500.0  
149258       47500.0  

[1082 rows x 9 columns]

In [26]:
maret2022[(maret2022.Location == "PONDOK INDAH MALL 2") & (maret2022.Product_ID == 67479)]

Tanggal  Product_ID                         Product  Location_ID  \
64      2022-02-28       67479  WARDAH E.SHADOW N.CLOR PSIONTE            1   
7943    2022-03-02       67479  WARDAH E.SHADOW N.CLOR PSIONTE            1   
82649   2022-03-20       67479  WARDAH E.SHADOW N.CLOR PSIONTE            1   
93645   2022-03-23       67479  WARDAH E.SHADOW N.CLOR PSIONTE            1   
113533  2022-03-27       67479  WARDAH E.SHADOW N.CLOR PSIONTE            1   
125064  2022-03-29       67479  WARDAH E.SHADOW N.CLOR PSIONTE            1   

                   Location  Quantity  Price_AfterPPN  Price_BeforePPN  \
64      PONDOK INDAH MALL 2         1         47500.0         43181.82   
7943    PONDOK INDAH MALL 2         2         95000.0         86363.64   
82649   PONDOK INDAH MALL 2         1         47500.0         43181.82   
93645   PONDOK INDAH MALL 2         2         95000.0         86363.64   
113533  PONDOK INDAH MALL 2         3        128250.0        116590.91   
125064  PONDOK INDAH MALL 2         1         47500.0         43181.82   

        Harga_Satuan  
64           47500.0  
7943         47500.0  
82649        47500.0  
93645        47500.0  
113533       42750.0  
125064       47500.0

In [27]:
maret2022.Location.value_counts()

SHOPEE WATSONS          6347
LAZADA MARGO CITY       3923
PONDOK INDAH MALL 2     3402
KOTA KASABLANKA         3091
SUMMARECON BEKASI       2763
                        ... 
TRANSMART MAGUWO         137
PONDOK INDAH STREET      133
DISCOVERY BALI            78
BALI SEMINYAK SQUARE      57
CLICK&COLLEC EXPRESS      16
Name: Location, Length: 169, dtype: int64

In [28]:
maret2022[(maret2022.Location == "SHOPEE WATSONS") & (maret2022.Product_ID == 55408)].Harga_Satuan.value_counts()

23200.000000    9
26100.000000    4
20300.000000    4
29000.000000    4
21750.000000    3
21355.491329    1
21802.941176    1
26611.764706    1
20726.470588    1
21618.181818    1
20867.391304    1
22942.222222    1
23010.869565    1
22415.816327    1
22888.888889    1
23634.146341    1
Name: Harga_Satuan, dtype: int64

In [29]:
temp  = maret2022[maret2022.Location == "SUMMARECON BEKASI"]

In [30]:
temp[temp.Product_ID == 10327].sort_values("Harga_Satuan")

Tanggal  Product_ID                         Product  Location_ID  \
61742  2022-03-14       10327  MAKE OVER P/STAY LIP STAIN A03           23   

                Location  Quantity  Price_AfterPPN  Price_BeforePPN  \
61742  SUMMARECON BEKASI         1        110000.0         100000.0   

       Harga_Satuan  
61742      110000.0

In [31]:
maret2022[maret2022.Product_ID == 10327].sort_values("Harga_Satuan")

Tanggal  Product_ID                         Product  Location_ID  \
21497   2022-03-04       10327  MAKE OVER P/STAY LIP STAIN A03          808   
17050   2022-03-03       10327  MAKE OVER P/STAY LIP STAIN A03          808   
149034  2022-04-03       10327  MAKE OVER P/STAY LIP STAIN A03          808   
145072  2022-04-02       10327  MAKE OVER P/STAY LIP STAIN A03          808   
117986  2022-03-27       10327  MAKE OVER P/STAY LIP STAIN A03          111   
...            ...         ...                             ...          ...   
67779   2022-03-16       10327  MAKE OVER P/STAY LIP STAIN A03           51   
136280  2022-04-01       10327  MAKE OVER P/STAY LIP STAIN A03           51   
137354  2022-04-01       10327  MAKE OVER P/STAY LIP STAIN A03          111   
139610  2022-04-02       10327  MAKE OVER P/STAY LIP STAIN A03            8   
141688  2022-04-02       10327  MAKE OVER P/STAY LIP STAIN A03           75   

                   Location  Quantity  Price_AfterPPN  Price_BeforePPN  \
21497        SHOPEE WATSONS         3        280500.0        255000.00   
17050        SHOPEE WATSONS         2        187000.0        170000.00   
149034       SHOPEE WATSONS         1         96050.0         86531.53   
145072       SHOPEE WATSONS         2        197750.0        178153.15   
117986       TRANS MAKASSAR         1         99000.0         90000.00   
...                     ...       ...             ...              ...   
67779   TANGERANG CITY MALL         1        110000.0        100000.00   
136280  TANGERANG CITY MALL         1        113000.0        101801.80   
137354       TRANS MAKASSAR         2        226000.0        203603.60   
139610      KOTA KASABLANKA         1        113000.0        101801.80   
141688            PASKAL 23         1        113000.0        101801.80   

        Harga_Satuan  
21497        93500.0  
17050        93500.0  
149034       96050.0  
145072       98875.0  
117986       99000.0  
...              ...  
67779       110000.0  
136280      113000.0  
137354      113000.0  
139610      113000.0  
141688      113000.0  

[65 rows x 9 columns]

<b> dari hasil data pada bulan maret diatas, dapat diasumsikan bahwa harga barang selalu berubah karena diskon ataupun potongan harga pada prodak </b>

# Melihat Hubungan Promo dengan Sell out

In [32]:
total_data = pd.concat([januari2021, februari2021, maret2021, april2021, mei2021,
                      juni2021, juli2021, agustus2021, september2021, oktober2021,
                      november2021, desember2021, januari2022, februari2022, maret2022,
                      april2022, mei2022]).reset_index()
total_data

index     Tanggal  Product_ID                         Product  \
0             0  2020-12-28     10568.0  EMINA CREAMYTINT 01 BRICK TOWN   
1             1  2020-12-28     15002.0   MO P/STAY TRANSPROOF LIPS CRM   
2             2  2020-12-28     15018.0  EMINA MAGIC POTION 04 SIENNA 5   
3             3  2020-12-28     15788.0  WARDAH COLORFIT VEL LIP/MAT 12   
4             4  2020-12-28     16288.0  WARDAH A/D PORE BLACKHEAD BALM   
...         ...         ...         ...                             ...   
2141166  151593  2022-05-29     68141.0          EMINA S/RUSH LIP SCRUB   
2141167  151594  2022-05-29     80428.0   EMINA DAILY MAT LOOSE PWDR 01   
2141168  151595  2022-05-29     80460.0   MO STICK MATTE LIP CRAYON 105   
2141169  151596  2022-05-29     85371.0  EMINA BRIGHT STUFF L/PWDR 55GR   
2141170  151597  2022-05-29     94754.0      WARDAH EYE LINER PCL BLACK   

         Location_ID             Location  Quantity  Price_AfterPPN  \
0                1.0  PONDOK INDAH MALL 2       1.0        44000.00   
1                1.0  PONDOK INDAH MALL 2       1.0       120000.00   
2                1.0  PONDOK INDAH MALL 2       1.0        36800.00   
3                1.0  PONDOK INDAH MALL 2       1.0        58400.00   
4                1.0  PONDOK INDAH MALL 2       1.0        33500.00   
...              ...                  ...       ...             ...   
2141166        917.0  ECOMM CASH DELIVERY       1.0        23460.00   
2141167        917.0  ECOMM CASH DELIVERY       1.0        24140.00   
2141168        917.0  ECOMM CASH DELIVERY       1.0        69429.00   
2141169        917.0  ECOMM CASH DELIVERY       4.0        54338.00   
2141170        917.0  ECOMM CASH DELIVERY       1.0        25690.99   

         Price_BeforePPN  Harga_Satuan  
0               40000.00           NaN  
1              109090.91           NaN  
2               33454.55           NaN  
3               53090.91           NaN  
4               30454.55           NaN  
...                  ...           ...  
2141166         21135.14           NaN  
2141167         21747.75           NaN  
2141168         62548.65           NaN  
2141169         48953.15           NaN  
2141170         23145.04           NaN  

[2141171 rows x 10 columns]

In [33]:
total_data.Location_ID.value_counts()

808.0    80385
807.0    49325
23.0     43185
8.0      42564
1.0      40721
         ...  
59.0        78
58.0        48
57.0        39
812.0       14
810.0        1
Name: Location_ID, Length: 182, dtype: int64

<b>untuk melihat hubungan promo dengan dengan sell out, mari melihat tiga tempat terbanyak hasil penjualannya</b>

In [34]:
## menambah suatu kolom baru berupa harga satuan untuk tinjauan
total_data["Harga_Satuan"] = total_data.Price_AfterPPN/ total_data.Quantity

In [35]:
total_data[total_data.Quantity <0]

index     Tanggal  Product_ID                         Product  \
51014     51014  2021-01-10     83552.0      MO P/STAY MAT PWDR FDT W22   
81012     81012  2021-01-20     15888.0  WARDAH C-DEFENSE SLEEPING MASK   
81201     81201  2021-01-20     15020.0  EMINA MAGIC POTION 06 SASSY 5.   
81204     81204  2021-01-20     18208.0  WARDAH UV SHIELD ESS SUNSCREEN   
81210     81210  2021-01-20     64713.0     MO BLUSH ON PROMISC PEACH03   
...         ...         ...         ...                             ...   
2115788  126215  2022-05-25     16883.0  WARDAH LIGHTENING BLUE CLAY MA   
2115809  126236  2022-05-25     54098.0  WARDAH C/S BRIGHT DAY CREAM 30   
2115816  126243  2022-05-25     63100.0   WARDAH E/DAY BB CRM NTRL 30ML   
2115821  126248  2022-05-25     80579.0  WARDAH E/X VOLUM XPERT MASCARA   
2115823  126250  2022-05-25     83881.0        WARDAH INSTA SKIN PRIMER   

         Location_ID            Location  Quantity  Price_AfterPPN  \
51014           29.0  CIBINONG CITY MALL      -1.0       -141000.0   
81012           52.0  BAYWALK MALL PLUIT      -1.0        -47000.0   
81201           78.0         FX SUDIRMAN      -1.0        -46000.0   
81204           78.0         FX SUDIRMAN      -1.0        -24500.0   
81210           78.0         FX SUDIRMAN      -1.0        -89000.0   
...              ...                 ...       ...             ...   
2115788        813.0           TOKOPEDIA      -1.0        -20000.0   
2115809        813.0           TOKOPEDIA      -1.0        -81700.0   
2115816        813.0           TOKOPEDIA      -1.0        -34600.0   
2115821        813.0           TOKOPEDIA      -1.0        -57300.0   
2115823        813.0           TOKOPEDIA      -1.0        -55300.0   

         Price_BeforePPN  Harga_Satuan  
51014         -128181.82      141000.0  
81012          -42727.27       47000.0  
81201          -41818.18       46000.0  
81204          -22272.73       24500.0  
81210          -80909.09       89000.0  
...                  ...           ...  
2115788        -18018.02       20000.0  
2115809        -73603.60       81700.0  
2115816        -31171.17       34600.0  
2115821        -51621.62       57300.0  
2115823        -49819.82       55300.0  

[968 rows x 10 columns]

In [36]:
total_data[total_data.Quantity == 0]

index     Tanggal  Product_ID                         Product  \
11413     11413  2020-12-30     10887.0   WARDAH COLORFIT ULTRA LIP/MAT   
20942     20942  2021-01-02     18238.0   MO P/STAY DM CC N30+MULTIFLIX   
23478     23478  2021-01-02     68936.0  WARDAH EXCLUSIVE MATTE L/CRM07   
32480     32480  2021-01-04     81134.0  WARDAH LIGHT BB CK/PWD 01 LIGH   
33558     33558  2021-01-05     54095.0     WARDAH W/SECRET EXFO LOTION   
...         ...         ...         ...                             ...   
2115737  126164  2022-05-25     83554.0      MO P/STAY MAT PWDR FDT W33   
2121258  131685  2022-05-26     64721.0       MO E/LINER PCL BLACK JACK   
2126933  137360  2022-05-27     94800.0       WARDAH EYE BROW PCL BROWN   
2133244  143671  2022-05-28     24889.0  EMINA D/MATE COMP PWDR 01 LIGH   
2133866  144293  2022-05-28     56755.0     EMINA BB CREAM CARAMEL 20ML   

         Location_ID              Location  Quantity  Price_AfterPPN  \
11413          174.0         GANDARIA CITY       0.0             0.0   
20942            8.0       KOTA KASABLANKA       0.0             0.0   
23478           96.0     TUNJUNGAN PLAZA 6       0.0             0.0   
32480          122.0       RING ROAD MEDAN       0.0             0.0   
33558            1.0   PONDOK INDAH MALL 2       0.0             0.0   
...              ...                   ...       ...             ...   
2115737        808.0        SHOPEE WATSONS       0.0          3200.0   
2121258        149.0   THEE MATIC MAJALAYA       0.0             0.0   
2126933        180.0        CITRA GARDEN 2       0.0             0.0   
2133244        162.0  THEPARK MAL SAWANGAN       0.0             0.0   
2133866        185.0    KOTA WISATA STREET       0.0             0.0   

         Price_BeforePPN  Harga_Satuan  
11413               0.00           NaN  
20942               0.00           NaN  
23478               0.00           NaN  
32480               0.00           NaN  
33558               0.00           NaN  
...                  ...           ...  
2115737          2882.88           inf  
2121258             0.00           NaN  
2126933             0.00           NaN  
2133244             0.00           NaN  
2133866             0.00           NaN  

[778 rows x 10 columns]

In [37]:
total_data[total_data.Quantity == 0].Quantity.value_counts()

0.0    778
Name: Quantity, dtype: int64

In [38]:
total_data[total_data.Price_AfterPPN ==0]

index     Tanggal  Product_ID                         Product  \
11413     11413  2020-12-30     10887.0   WARDAH COLORFIT ULTRA LIP/MAT   
20942     20942  2021-01-02     18238.0   MO P/STAY DM CC N30+MULTIFLIX   
23478     23478  2021-01-02     68936.0  WARDAH EXCLUSIVE MATTE L/CRM07   
32480     32480  2021-01-04     81134.0  WARDAH LIGHT BB CK/PWD 01 LIGH   
33558     33558  2021-01-05     54095.0     WARDAH W/SECRET EXFO LOTION   
...         ...         ...         ...                             ...   
2115721  126148  2022-05-25     80742.0   EMINA MP A/SOL MOIST GEL 20ML   
2121258  131685  2022-05-26     64721.0       MO E/LINER PCL BLACK JACK   
2126933  137360  2022-05-27     94800.0       WARDAH EYE BROW PCL BROWN   
2133244  143671  2022-05-28     24889.0  EMINA D/MATE COMP PWDR 01 LIGH   
2133866  144293  2022-05-28     56755.0     EMINA BB CREAM CARAMEL 20ML   

         Location_ID              Location  Quantity  Price_AfterPPN  \
11413          174.0         GANDARIA CITY       0.0             0.0   
20942            8.0       KOTA KASABLANKA       0.0             0.0   
23478           96.0     TUNJUNGAN PLAZA 6       0.0             0.0   
32480          122.0       RING ROAD MEDAN       0.0             0.0   
33558            1.0   PONDOK INDAH MALL 2       0.0             0.0   
...              ...                   ...       ...             ...   
2115721        808.0        SHOPEE WATSONS       0.0             0.0   
2121258        149.0   THEE MATIC MAJALAYA       0.0             0.0   
2126933        180.0        CITRA GARDEN 2       0.0             0.0   
2133244        162.0  THEPARK MAL SAWANGAN       0.0             0.0   
2133866        185.0    KOTA WISATA STREET       0.0             0.0   

         Price_BeforePPN  Harga_Satuan  
11413                0.0           NaN  
20942                0.0           NaN  
23478                0.0           NaN  
32480                0.0           NaN  
33558                0.0           NaN  
...                  ...           ...  
2115721              0.0           NaN  
2121258              0.0           NaN  
2126933              0.0           NaN  
2133244              0.0           NaN  
2133866              0.0           NaN  

[936 rows x 10 columns]

In [39]:
total_data[total_data.Price_AfterPPN ==0].Quantity.value_counts()

0.0     633
1.0     234
2.0      42
3.0      10
4.0       6
11.0      3
5.0       3
6.0       2
12.0      1
39.0      1
7.0       1
Name: Quantity, dtype: int64

<b> terdapat data dengan Quantity == 0 yaitu 778 data dan dengan harga penjulaan sebesar 0 yaitu ada 936 data</b> karena 936 data dari 2141171 data hanya 0.0437% akhiranya diputuskan untuk di drop

In [40]:
total_data = total_data.drop(total_data[total_data.Price_AfterPPN == 0].index)
total_data = total_data.drop(total_data[total_data.Quantity == 0].index)
total_data

index     Tanggal  Product_ID                         Product  \
0             0  2020-12-28     10568.0  EMINA CREAMYTINT 01 BRICK TOWN   
1             1  2020-12-28     15002.0   MO P/STAY TRANSPROOF LIPS CRM   
2             2  2020-12-28     15018.0  EMINA MAGIC POTION 04 SIENNA 5   
3             3  2020-12-28     15788.0  WARDAH COLORFIT VEL LIP/MAT 12   
4             4  2020-12-28     16288.0  WARDAH A/D PORE BLACKHEAD BALM   
...         ...         ...         ...                             ...   
2141166  151593  2022-05-29     68141.0          EMINA S/RUSH LIP SCRUB   
2141167  151594  2022-05-29     80428.0   EMINA DAILY MAT LOOSE PWDR 01   
2141168  151595  2022-05-29     80460.0   MO STICK MATTE LIP CRAYON 105   
2141169  151596  2022-05-29     85371.0  EMINA BRIGHT STUFF L/PWDR 55GR   
2141170  151597  2022-05-29     94754.0      WARDAH EYE LINER PCL BLACK   

         Location_ID             Location  Quantity  Price_AfterPPN  \
0                1.0  PONDOK INDAH MALL 2       1.0        44000.00   
1                1.0  PONDOK INDAH MALL 2       1.0       120000.00   
2                1.0  PONDOK INDAH MALL 2       1.0        36800.00   
3                1.0  PONDOK INDAH MALL 2       1.0        58400.00   
4                1.0  PONDOK INDAH MALL 2       1.0        33500.00   
...              ...                  ...       ...             ...   
2141166        917.0  ECOMM CASH DELIVERY       1.0        23460.00   
2141167        917.0  ECOMM CASH DELIVERY       1.0        24140.00   
2141168        917.0  ECOMM CASH DELIVERY       1.0        69429.00   
2141169        917.0  ECOMM CASH DELIVERY       4.0        54338.00   
2141170        917.0  ECOMM CASH DELIVERY       1.0        25690.99   

         Price_BeforePPN  Harga_Satuan  
0               40000.00      44000.00  
1              109090.91     120000.00  
2               33454.55      36800.00  
3               53090.91      58400.00  
4               30454.55      33500.00  
...                  ...           ...  
2141166         21135.14      23460.00  
2141167         21747.75      24140.00  
2141168         62548.65      69429.00  
2141169         48953.15      13584.50  
2141170         23145.04      25690.99  

[2140090 rows x 10 columns]

## untuk daerah dengan location_id 808

In [41]:
temp = total_data[total_data.Location_ID == 808]
temp

index     Tanggal  Product_ID                         Product  \
3515       3515  2020-12-28     11543.0   Crystallure Supr Revital Rich   
3516       3516  2020-12-28     11551.0  Crystallure Act Booster Essenc   
3517       3517  2020-12-28     15469.0  WARDAH NAT.D MILKY OAT EXF SCR   
3518       3518  2020-12-28     31083.0  WARDAH SCENT PEACEFUL EDT 35ML   
3519       3519  2020-12-28     31085.0  WARDAH SCENT BLISS B.MIST100ML   
...         ...         ...         ...                             ...   
2141074  151501  2022-05-29     85201.0   WARDAH EXC MATTE LIPS 12 3.5G   
2141075  151502  2022-05-29     85371.0  EMINA BRIGHT STUFF L/PWDR 55GR   
2141076  151503  2022-05-29     94757.0             WARDAH EYE SHADOW A   
2141077  151504  2022-05-29     96969.0  WARDAH LIGHT PWDR FDT LF S/PIN   
2141078  151505  2022-05-29     96972.0     WARDAH TWC REF BEIGE 02 12G   

         Location_ID        Location  Quantity  Price_AfterPPN  \
3515           808.0  SHOPEE WATSONS       1.0        188900.0   
3516           808.0  SHOPEE WATSONS       1.0        215900.0   
3517           808.0  SHOPEE WATSONS       2.0         31000.0   
3518           808.0  SHOPEE WATSONS       3.0        131700.0   
3519           808.0  SHOPEE WATSONS       7.0        206500.0   
...              ...             ...       ...             ...   
2141074        808.0  SHOPEE WATSONS       2.0         51600.0   
2141075        808.0  SHOPEE WATSONS       3.0         55200.0   
2141076        808.0  SHOPEE WATSONS       1.0         48500.0   
2141077        808.0  SHOPEE WATSONS       1.0         35500.0   
2141078        808.0  SHOPEE WATSONS       1.0         34500.0   

         Price_BeforePPN  Harga_Satuan  
3515           171727.27      188900.0  
3516           196272.73      215900.0  
3517            28181.82       15500.0  
3518           119727.27       43900.0  
3519           187727.27       29500.0  
...                  ...           ...  
2141074         46486.49       25800.0  
2141075         49729.73       18400.0  
2141076         43693.69       48500.0  
2141077         31981.98       35500.0  
2141078         31081.08       34500.0  

[80292 rows x 10 columns]

<b> shopee adalah salah satu platform yang memiliki banyak promo. kita akan ambil dua prodak dengan penjualan terbanyak </b> hal ini agar data yang dapat kita tinjau banyak

In [42]:
temp.Product_ID.value_counts()

55408.0    479
82983.0    455
50337.0    451
50022.0    435
80580.0    431
          ... 
24901.0      1
63991.0      1
69947.0      1
19946.0      1
24591.0      1
Name: Product_ID, Length: 1108, dtype: int64

### Shopee dengan Product_ID == 55408

In [43]:
temp[temp.Product_ID == 55408]

index     Tanggal  Product_ID                         Product  \
3546       3546  2020-12-28     55408.0  EMINA SUN PROTECTION SPF30 60M   
7364       7364  2020-12-29     55408.0  EMINA SUN PROTECTION SPF30 60M   
11541     11541  2020-12-30     55408.0  EMINA SUN PROTECTION SPF30 60M   
15842     15842  2020-12-31     55408.0  EMINA SUN PROTECTION SPF30 60M   
20627     20627  2021-01-01     55408.0  EMINA SUN PROTECTION SPF30 60M   
...         ...         ...         ...                             ...   
2111511  121938  2022-05-24     55408.0  EMINA SUN BATTLE SPF 30+++ 60M   
2122449  132876  2022-05-26     55408.0  EMINA SUN BATTLE SPF 30+++ 60M   
2127399  137826  2022-05-27     55408.0  EMINA SUN BATTLE SPF 30+++ 60M   
2134225  144652  2022-05-28     55408.0  EMINA SUN BATTLE SPF 30+++ 60M   
2141016  151443  2022-05-29     55408.0  EMINA SUN BATTLE SPF 30+++ 60M   

         Location_ID        Location  Quantity  Price_AfterPPN  \
3546           808.0  SHOPEE WATSONS     241.0       4530800.0   
7364           808.0  SHOPEE WATSONS      78.0       1466400.0   
11541          808.0  SHOPEE WATSONS      40.0        752000.0   
15842          808.0  SHOPEE WATSONS      20.0        376000.0   
20627          808.0  SHOPEE WATSONS       1.0         18800.0   
...              ...             ...       ...             ...   
2111511        808.0  SHOPEE WATSONS     -12.0       -155150.0   
2122449        808.0  SHOPEE WATSONS      23.0        667000.0   
2127399        808.0  SHOPEE WATSONS      13.0        377000.0   
2134225        808.0  SHOPEE WATSONS       7.0        203000.0   
2141016        808.0  SHOPEE WATSONS      10.0        290000.0   

         Price_BeforePPN  Harga_Satuan  
3546          4118909.09  18800.000000  
7364          1333090.91  18800.000000  
11541          683636.36  18800.000000  
15842          341818.18  18800.000000  
20627           17090.91  18800.000000  
...                  ...           ...  
2111511       -139774.77  12929.166667  
2122449        600900.90  29000.000000  
2127399        339639.64  29000.000000  
2134225        182882.88  29000.000000  
2141016        261261.26  29000.000000  

[479 rows x 10 columns]

In [44]:
temp[temp.Product_ID == 55408].Harga_Satuan

3546       18800.000000
7364       18800.000000
11541      18800.000000
15842      18800.000000
20627      18800.000000
               ...     
2111511    12929.166667
2122449    29000.000000
2127399    29000.000000
2134225    29000.000000
2141016    29000.000000
Name: Harga_Satuan, Length: 479, dtype: float64

# Checking Data Watson untuk bulan April 2022

In [45]:
april2022[april2022.Product_ID == 10568]

Tanggal  Product_ID                         Product  Location_ID  \
0       2022-04-04       10568  EMINA CREAMYTINT 01 BRICK TOWN            1   
185     2022-04-04       10568  EMINA CREAMYTINT 01 BRICK TOWN           12   
298     2022-04-04       10568  EMINA CREAMYTINT 01 BRICK TOWN           23   
368     2022-04-04       10568  EMINA CREAMYTINT 01 BRICK TOWN           26   
637     2022-04-04       10568  EMINA CREAMYTINT 01 BRICK TOWN           51   
...            ...         ...                             ...          ...   
157365  2022-05-01       10568  EMINA CREAMYTINT 01 BRICK TOWN          181   
157510  2022-05-01       10568  EMINA CREAMYTINT 01 BRICK TOWN          183   
157654  2022-05-01       10568  EMINA CREAMYTINT 01 BRICK TOWN          185   
157799  2022-05-01       10568  EMINA CREAMYTINT 01 BRICK TOWN          190   
158077  2022-05-01       10568  EMINA CREAMYTINT 01 BRICK TOWN          813   

                   Location  Quantity  Price_AfterPPN  Price_BeforePPN  
0       PONDOK INDAH MALL 2         1         45000.0         40540.54  
185             KRAMAT JATI         1         45000.0         40540.54  
298       SUMMARECON BEKASI         1         45000.0         40540.54  
368           MAL CIJANTUNG         1         45000.0         40540.54  
637     TANGERANG CITY MALL         1         45000.0         40540.54  
...                     ...       ...             ...              ...  
157365   GRAGE MALL CIREBON         3        135000.0        121621.62  
157510    MARGO CITY REBORN         1         45000.0         40540.54  
157654   KOTA WISATA STREET         2         90000.0         81081.08  
157799    TEBET RAYA STREET         1         45000.0         40540.54  
158077            TOKOPEDIA         1         31500.0         28378.38  

[652 rows x 8 columns]

In [46]:
april2022[(april2022.Location_ID == 29) & (april2022.Product_ID == 10568)]

Tanggal  Product_ID                         Product  Location_ID  \
63039   2022-04-20       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   
79309   2022-04-23       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   
90930   2022-04-24       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   
101026  2022-04-25       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   
112985  2022-04-27       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   
119915  2022-04-28       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   
128470  2022-04-29       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   
139062  2022-04-30       10568  EMINA CREAMYTINT 01 BRICK TOWN           29   

                  Location  Quantity  Price_AfterPPN  Price_BeforePPN  
63039   CIBINONG CITY MALL         1         45000.0         40540.54  
79309   CIBINONG CITY MALL         1         36000.0         32432.43  
90930   CIBINONG CITY MALL         1         36000.0         32432.43  
101026  CIBINONG CITY MALL         1         45000.0         40540.54  
112985  CIBINONG CITY MALL         1         45000.0         40540.54  
119915  CIBINONG CITY MALL         1         45000.0         40540.54  
128470  CIBINONG CITY MALL         1         45000.0         40540.54  
139062  CIBINONG CITY MALL         4        180000.0        162162.16

In [47]:
april2022[(april2022.Location_ID == 29) & (april2022.Product_ID == 10568)].Price_AfterPPN.sum()

477000.0

In [48]:
april2022[(april2022.Location_ID == 29) & (april2022.Product_ID == 10568)].Quantity.sum()

11

In [49]:
total_data.Location_ID.max()

919.0

In [50]:
temp = pd.read_csv("BiVi\Watson_Cibinong_Mall\Oktober_2021.csv")

In [51]:
temp

series  default_code  \
0       BARE WITH ME          1840   
1       BEAUTY BLISS           773   
2       BEAUTY BLISS           774   
3       BEAUTY BLISS          1014   
4       BRIGHT STUFF          1880   
..               ...           ...   
348     BARE WITH ME          1888   
349        EXCLUSIVE          2037   
350     BRIGHT STUFF          1993   
351         COLORFIT          2116   
352  NO SERIES EMINA           814   

                                          product_name     price  qty_week1  \
0    Emina Bare With Me Mineral Cushion 02 Natural ...  126000.0          0   
1      Emina Beauty Bliss BB Cream Light 20 ml (Pc(s))   34000.0          0   
2    Emina Beauty Bliss BB Cream Natural 20 ml (Pc(s))   34000.0          0   
3    Emina Beauty Bliss BB Cream Caramel 20 ml (Pc(s))   34000.0          0   
4          Emina Bright Stuff Moisturizing Cream 20 ml   23500.0          0   
..                                                 ...       ...        ...   
348  Emina Bare With Me Mineral Loose Powder 01 Fai...       NaN          0   
349  Wardah Exclusive Liquid Foundation 04 Natural ...       NaN          0   
350         Emina Bright Stuff Essence Sheet Mask 23 g       NaN          0   
351  Wardah Colorfit Velvet Matte Lip Mousse 05 Art...       NaN          0   
352  Emina Pop Rouge Pressed Eye Shadow Nude 3.3 gr...       NaN          0   

     qty_week2  qty_week3  qty_week4  qty_week5  total_qty_sellin  ...  \
0            0          3          0          0                 3  ...   
1            0          4          0          0                 4  ...   
2            0          1          0          0                 1  ...   
3            0          0          0          0                 0  ...   
4            0         12          0          0                12  ...   
..         ...        ...        ...        ...               ...  ...   
348          0          2          0          0                 2  ...   
349          0          1          0          0                 1  ...   
350          0          5          0          0                 5  ...   
351          0          4          0          0                 4  ...   
352          0          1          0          0                 1  ...   

     qty_day24  qty_day25  qty_day26  qty_day27  qty_day28  qty_day29  \
0          0.0        0.0        0.0        0.0        0.0        0.0   
1          0.0        0.0        0.0        0.0        0.0        0.0   
2          0.0        0.0        0.0        0.0        0.0        0.0   
3          0.0        0.0        0.0        0.0        0.0        0.0   
4          0.0        0.0        0.0        0.0        0.0        0.0   
..         ...        ...        ...        ...        ...        ...   
348        NaN        NaN        NaN        NaN        NaN        NaN   
349        NaN        NaN        NaN        NaN        NaN        NaN   
350        NaN        NaN        NaN        NaN        NaN        NaN   
351        NaN        NaN        NaN        NaN        NaN        NaN   
352        NaN        NaN        NaN        NaN        NaN        NaN   

     qty_day30  qty_day31  total_qty_sellout  total_achievement_sellout  
0          0.0        0.0                2.0                   252000.0  
1          0.0        0.0                3.0                   102000.0  
2          0.0        0.0                2.0                    68000.0  
3          0.0        0.0                1.0                    34000.0  
4          0.0        0.0                3.0                    70500.0  
..         ...        ...                ...                        ...  
348        NaN        NaN                NaN                        NaN  
349        NaN        NaN                NaN                        NaN  
350        NaN        NaN                NaN                        NaN  
351        NaN        NaN                NaN                        NaN  
352        NaN        NaN   

<b> untuk dapat melakukan scripting ke data watson hasil BiVi dan data sistem watson, mesti ada primary key </b>

In [52]:
oktober2021[(oktober2021.Location_ID == 29)].Product.value_counts()

WARDAH COLORFIT LAD LIP/PAINT     25
EMINA LIP GLOSSY STAIN 01 AUTU    19
WARDAH UV SHIELD ESS SUNSCREEN    19
MO P/STAY TRANSPROOF LIPS CRM     15
WARDAH UV SHIELD A/F ESSENCE S    15
                                  ..
MO P/STAY W/ LIQ FDT W22 33ML      1
EMINA MAGIC POTION 03 SMITTEN      1
WARDAH EXC FLAWLESS CC 02          1
WARDAH EYE BROW PCL BLACK          1
EMINA CREAMATTE 07 TUMBLEWEED      1
Name: Product, Length: 363, dtype: int64

In [53]:
substrings = ['EMINA BARE WITH ME']
regstr = '|'.join(substrings)
november2021[(november2021.Product.str.upper().str.contains(regstr))]

Tanggal  Product_ID                         Product  Location_ID  \
166     2021-10-25       17752  EMINA BARE WITH ME M/ FDT 01 L            8   
382     2021-10-25       17753  EMINA BARE WITH ME M/ FDT 02 N           22   
432     2021-10-25       17752  EMINA BARE WITH ME M/ FDT 01 L           23   
1645    2021-10-25       17753  EMINA BARE WITH ME M/ FDT 02 N          114   
2710    2021-10-25       17752  EMINA BARE WITH ME M/ FDT 01 L          808   
...            ...         ...                             ...          ...   
115329  2021-11-21       17752  EMINA BARE WITH ME M/ FDT 01 L          119   
115470  2021-11-21       17753  EMINA BARE WITH ME M/ FDT 02 N          122   
116261  2021-11-21       17753  EMINA BARE WITH ME M/ FDT 02 N          147   
116444  2021-11-21       17754  EMINA BARE WITH ME M/ FDT 03 C          152   
116993  2021-11-21       17755  EMINA BARE WITH ME M/ FDT 04 M          181   

                  Location  Quantity  Price_AfterPPN  Price_BeforePPN  
166        KOTA KASABLANKA         1         79000.0         71818.18  
382      PLAZA PONDOK GEDE         1         79000.0         71818.18  
432      SUMMARECON BEKASI         1         79000.0         71818.18  
1645        DISCOVERY BALI         1         79000.0         71818.18  
2710        SHOPEE WATSONS         1         63200.0         57454.55  
...                    ...       ...             ...              ...  
115329       KINGS BANDUNG         1         79000.0         71818.18  
115470     RING ROAD MEDAN         1         79000.0         71818.18  
116261       SKA PEKANBARU         1         79000.0         71818.18  
116444  EKA LOKASARI BOGOR         1         79000.0         71818.18  
116993  GRAGE MALL CIREBON         1         79000.0         71818.18  

[112 rows x 8 columns]

In [54]:
november2021[november2021.Product_ID == 16882]

Tanggal  Product_ID                         Product  Location_ID  \
1929    2021-10-25       16882  WARDAH LIGHTENING GENTLE EXFOL          132   
2019    2021-10-25       16882  WARDAH LIGHTENING GENTLE EXFOL          138   
2508    2021-10-25       16882  WARDAH LIGHTENING GENTLE EXFOL          178   
3708    2021-10-26       16882  WARDAH LIGHTENING GENTLE EXFOL           33   
4483    2021-10-26       16882  WARDAH LIGHTENING GENTLE EXFOL           93   
...            ...         ...                             ...          ...   
115595  2021-11-21       16882  WARDAH LIGHTENING GENTLE EXFOL          126   
115720  2021-11-21       16882  WARDAH LIGHTENING GENTLE EXFOL          129   
116563  2021-11-21       16882  WARDAH LIGHTENING GENTLE EXFOL          157   
116616  2021-11-21       16882  WARDAH LIGHTENING GENTLE EXFOL          160   
116697  2021-11-21       16882  WARDAH LIGHTENING GENTLE EXFOL          166   

                    Location  Quantity  Price_AfterPPN  Price_BeforePPN  
1929    PALEMBANG INDAH MALL         1         16000.0         14545.45  
2019         AYANI PONTIANAK         1         16000.0         14545.45  
2508       METROPOLITAN MALL         1         16000.0         14545.45  
3708           BOTANI SQUARE         1         16000.0         14545.45  
4483    LIV PLAZA BALIKPAPAN         1         16000.0         14545.45  
...                      ...       ...             ...              ...  
115595     GRESS MALL GRESIK         1         16000.0         14545.45  
115720         LEVEL 21 BALI         1         16000.0         14545.45  
116563      PALEMBANG SQUARE         1         16000.0         14545.45  
116616        PALEMBANG ICON         1         16000.0         14545.45  
116697  CITRA RAYA TANGERANG         1         16000.0         14545.45  

[214 rows x 8 columns]

In [55]:
temp[temp.default_code == 1840]

series  default_code  \
0  BARE WITH ME          1840   

                                        product_name     price  qty_week1  \
0  Emina Bare With Me Mineral Cushion 02 Natural ...  126000.0          0   

   qty_week2  qty_week3  qty_week4  qty_week5  total_qty_sellin  ...  \
0          0          3          0          0                 3  ...   

   qty_day24  qty_day25  qty_day26  qty_day27  qty_day28  qty_day29  \
0        0.0        0.0        0.0        0.0        0.0        0.0   

   qty_day30  qty_day31  total_qty_sellout  total_achievement_sellout  
0        0.0        0.0                2.0                   252000.0  

[1 rows x 44 columns]

In [56]:
total_data[total_data.Product_ID == 73746]

index     Tanggal  Product_ID                       Product  \
1933       1933  2020-12-28     73746.0  WARDAH SHP NUTRI SHINE 170ML   
3639       3639  2020-12-28     73746.0  WARDAH SHP NUTRI SHINE 170ML   
7490       7490  2020-12-30     73746.0  WARDAH SHP NUTRI SHINE 170ML   
11474     11474  2020-12-30     73746.0  WARDAH SHP NUTRI SHINE 170ML   
11565     11565  2020-12-30     73746.0  WARDAH SHP NUTRI SHINE 170ML   
...         ...         ...         ...                           ...   
2120499  130926  2022-05-26     73746.0  WARDAH SHP NUTRI SHINE 170ML   
2122257  132684  2022-05-26     73746.0  WARDAH SHP NUTRI SHINE 170ML   
2123718  134145  2022-05-27     73746.0  WARDAH SHP NUTRI SHINE 170ML   
2134509  144936  2022-05-29     73746.0  WARDAH SHP NUTRI SHINE 170ML   
2138954  149381  2022-05-29     73746.0  WARDAH SHP NUTRI SHINE 170ML   

         Location_ID              Location  Quantity  Price_AfterPPN  \
1933           104.0      ICON MALL GRESIK       2.0         44000.0   
3639           813.0             TOKOPEDIA       2.0         38500.0   
7490             1.0   PONDOK INDAH MALL 2       1.0         22000.0   
11474          176.0  BINTARO STREET STORE       1.0         22000.0   
11565          808.0        SHOPEE WATSONS       1.0         22000.0   
...              ...                   ...       ...             ...   
2120499        129.0         LEVEL 21 BALI       1.0         25300.0   
2122257        190.0     TEBET RAYA STREET       1.0         25300.0   
2123718         31.0     GRAND GALAXY PARK       1.0         25300.0   
2134509          1.0   PONDOK INDAH MALL 2       1.0         25300.0   
2138954        128.0     BINJAI MALL MEDAN       2.0         50600.0   

         Price_BeforePPN  Harga_Satuan  
1933            40000.00       22000.0  
3639            35000.00       19250.0  
7490            20000.00       22000.0  
11474           20000.00       22000.0  
11565           20000.00       22000.0  
...                  ...           ...  
2120499         22792.79       25300.0  
2122257         22792.79       25300.0  
2123718         22792.79       25300.0  
2134509         22792.79       25300.0  
2138954         45585.59       25300.0  

[1769 rows x 10 columns]

In [57]:
total_data.Location_ID.max()

919.0

# Mapping Sehingga Code Product Watson menggunakan Odoo Code

In [58]:
table_mapping = pd.read_csv("Data_Mapping/Product_Mapping.csv")
table_mapping

PLU        Barcode  ERP Product Code  \
0    73746.0  8993137698443              1486   
1    73744.0  8993137698450              4576   
2    73745.0  8993137698467              1488   
3    73743.0  8993137698436              1485   
4    31084.0  8993137698429              1884   
..       ...            ...               ...   
219      NaN  8993137720823              4544   
220      NaN  8993137721035              4551   
221      NaN  8993137713986              3112   
222      NaN  8993137713207              3748   
223      NaN  8993137723756              3893   

                                              ERP Name  
0            Wardah Nutri Shine Shampoo 170 ml (Pc(s))  
1     Wardah Hairfall Treatment Shampoo 170 ml (Pc(s))  
2            Wardah Daily Fresh Shampoo 170 ml (Pc(s))  
3          Wardah Anti Dandruff Shampoo 170 ml (Pc(s))  
4    Wardah Scentsation Purity Eau De Toilette 35 m...  
..                                                 ...  
219            Kahf True Brotherhood Essential Package  
220             Kahf Skin Protection Essential Package  
221              Wardah Bundling White Secret - Watson  
222  LABORE Sensitive Skin Care GentleBiome Mild Cl...  
223  Kahf Triple Action Oil and Comedo Defense Face...  

[224 rows x 4 columns]

In [59]:
for i in table_mapping.PLU:
    print(i)

73746.0
73744.0
73745.0
73743.0
31084.0
31089.0
31083.0
31088.0
31081.0
31087.0
31080.0
31086.0
31077.0
31085.0
80486.0
80431.0
80430.0
32080.0
33009.0
55271.0
54094.0
54097.0
55366.0
54099.0
54095.0
54098.0
55367.0
81240.0
57663.0
50332.0
50333.0
80581.0
80580.0
57872.0
57870.0
57871.0
47735.0
66203.0
50341.0
66204.0
81241.0
82921.0
82920.0
68151.0
64524.0
68150.0
47734.0
58475.0
58473.0
58474.0
62516.0
47732.0
51724.0
50801.0
50803.0
50802.0
50800.0
50799.0
57662.0
50331.0
50334.0
50340.0
51725.0
63096.0
63098.0
63097.0
67159.0
57824.0
57825.0
57925.0
57826.0
67160.0
63100.0
63099.0
63101.0
50343.0
50345.0
50337.0
50338.0
55076.0
55270.0
68541.0
56805.0
57062.0
57873.0
50335.0
57823.0
47712.0
47711.0
47710.0
47708.0
67335.0
67334.0
51727.0
51141.0
11540.0
11543.0
11544.0
11551.0
11552.0
11554.0
11562.0
15117.0
15120.0
14649.0
14647.0
15114.0
15115.0
15116.0
15888.0
15466.0
15467.0
15468.0
15469.0
15470.0
16257.0
16285.0
16346.0
16286.0
16287.0
16288.0
16882.0
16883.0
16884.0
16885.0


In [60]:
total_data[total_data.Product_ID == table_mapping.PLU[0]].Product_ID = table_mapping["ERP Product Code"][0]

C:\Users\muham\AppData\Local\Temp\ipykernel_20700\22956139.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  total_data[total_data.Product_ID == table_mapping.PLU[0]].Product_ID = table_mapping["ERP Product Code"][0]


In [61]:
total_data[total_data.Product_ID == table_mapping["ERP Product Code"][0]]

Empty DataFrame
Columns: [index, Tanggal, Product_ID, Product, Location_ID, Location, Quantity, Price_AfterPPN, Price_BeforePPN, Harga_Satuan]
Index: []

In [62]:
total_data

index     Tanggal  Product_ID                         Product  \
0             0  2020-12-28     10568.0  EMINA CREAMYTINT 01 BRICK TOWN   
1             1  2020-12-28     15002.0   MO P/STAY TRANSPROOF LIPS CRM   
2             2  2020-12-28     15018.0  EMINA MAGIC POTION 04 SIENNA 5   
3             3  2020-12-28     15788.0  WARDAH COLORFIT VEL LIP/MAT 12   
4             4  2020-12-28     16288.0  WARDAH A/D PORE BLACKHEAD BALM   
...         ...         ...         ...                             ...   
2141166  151593  2022-05-29     68141.0          EMINA S/RUSH LIP SCRUB   
2141167  151594  2022-05-29     80428.0   EMINA DAILY MAT LOOSE PWDR 01   
2141168  151595  2022-05-29     80460.0   MO STICK MATTE LIP CRAYON 105   
2141169  151596  2022-05-29     85371.0  EMINA BRIGHT STUFF L/PWDR 55GR   
2141170  151597  2022-05-29     94754.0      WARDAH EYE LINER PCL BLACK   

         Location_ID             Location  Quantity  Price_AfterPPN  \
0                1.0  PONDOK INDAH MALL 2       1.0        44000.00   
1                1.0  PONDOK INDAH MALL 2       1.0       120000.00   
2                1.0  PONDOK INDAH MALL 2       1.0        36800.00   
3                1.0  PONDOK INDAH MALL 2       1.0        58400.00   
4                1.0  PONDOK INDAH MALL 2       1.0        33500.00   
...              ...                  ...       ...             ...   
2141166        917.0  ECOMM CASH DELIVERY       1.0        23460.00   
2141167        917.0  ECOMM CASH DELIVERY       1.0        24140.00   
2141168        917.0  ECOMM CASH DELIVERY       1.0        69429.00   
2141169        917.0  ECOMM CASH DELIVERY       4.0        54338.00   
2141170        917.0  ECOMM CASH DELIVERY       1.0        25690.99   

         Price_BeforePPN  Harga_Satuan  
0               40000.00      44000.00  
1              109090.91     120000.00  
2               33454.55      36800.00  
3               53090.91      58400.00  
4               30454.55      33500.00  
...                  ...           ...  
2141166         21135.14      23460.00  
2141167         21747.75      24140.00  
2141168         62548.65      69429.00  
2141169         48953.15      13584.50  
2141170         23145.04      25690.99  

[2140090 rows x 10 columns]

In [63]:
len(total_data.Product_ID.value_counts())

1485

In [64]:
len(data.location.value_counts())

NameError: name 'data' is not defined

In [65]:
len(total_data.Location.value_counts())

182

In [66]:
len(data.product_id.value_counts())

NameError: name 'data' is not defined